core/src/main/scala/org/http4s/Message.scala

```diff
@@ -224,6 +225,53 @@ object Request {
   }
 }
 
+/**
+ * Represents that a service either returns a [[Response]] or a [[Pass]] to fall through
+ * to another service.
+ */
+sealed trait MaybeResponse {
```

Comment:
Looks like there are a number of places that would be well served by a `notReallyAFlatMapButFeelsKindOfLikeOne(f: Response => MaybeResponse)` method. Does that form of method have a common name?